solid/src/test/java/com/inrupt/client/solid/SolidClientTest.java

```diff
@@ -379,28 +380,286 @@ private static Stream<Arguments> testExceptionalResources() {
+            final ProblemDetails problemDetails
+    ) {
+        final Headers headers = Headers.of(Collections.singletonMap("x-key", Arrays.asList("value")));
+        final SolidClient solidClient = new SolidClient(ClientProvider.getClient(), headers, false);
+        final SolidContainer resource = new SolidContainer(URI.create("http://example.com"));
+
         final SolidClientException exception = assertThrows(
                 clazz,
                 () -> solidClient.handleResponse(resource, headers, "message")
                         .apply(new Response<byte[]>() {
                             @Override
                             public byte[] body() {
-                                return new byte[0];
+                                final ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                                try {
+                                    jsonService.toJson(problemDetails, bos);
+                                } catch (IOException e) {
+                                    throw new RuntimeException(e);
+                                }
+                                return bos.toByteArray();
                             }
 
                             @Override
                             public Headers headers() {
+                                final List<String> headerValues = new ArrayList<>();
+                                headerValues.add("application/problem+json");
+                                final Map<String, List<String>> headerMap = new HashMap<>();
+                                headerMap.put("Content-Type", headerValues);
+                                return Headers.of(headerMap);
+                            }
+
+                            @Override
+                            public URI uri() {
                                 return null;
                             }
 
+                            @Override
+                            public int statusCode() {
+                                return problemDetails.getStatus();
+                            }
+                        })
+        );
+        assertEquals(problemDetails.getStatus(), exception.getStatusCode());
+        assertEquals(problemDetails.getType(), exception.getProblemDetails().getType());
+        assertEquals(problemDetails.getTitle(), exception.getProblemDetails().getTitle());
+        assertEquals(problemDetails.getStatus(), exception.getProblemDetails().getStatus());
+        assertEquals(problemDetails.getDetails(), exception.getProblemDetails().getDetails());
+        assertEquals(problemDetails.getInstance(), exception.getProblemDetails().getInstance());
+    }
+
+    private static Stream<Arguments> testRfc9457Exceptions() {
+        return Stream.of(
+                Arguments.of(
+                        BadRequestException.class,
+                        new ProblemDetails(
+                                URI.create("https://example.org/type"),
```

Comment:
I would extract what I can into locals of `testRfc9457Exceptions`.
Anything to make the test case rows more readable.

Reply:
Attempted improvement in 8eed2118b465f22d61cb91b6c9c0e0b1be1247a3

Reply:
Looks good.